### ISSUE TITLE
Untrusted code may only update documents by ID when calling `documents.insert`

### ISSUE THREAD

**Issue by make-github-pseudonymous-again** [infoderm/patients#228]
Getting this error when uploading documents. The upload appears to work though. Seems to be related to a recent commit making the `documents.insert` method async.

```
Uncaught (in promise) 
{…}
​
details: undefined
​
error: 403
​
errorType: "Meteor.Error"
​
isClientSafe: true
​
message: "Not permitted. Untrusted code may only update documents by ID. [403]"
​
reason: "Not permitted. Untrusted code may only update documents by ID."
​
stack: "errorClass@http://localhost:12345/packages/meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:689:20\nthrowIfSelectorIsNotId@http://localhost:12345/packages/allow-deny.js?hash=000d79baa5a670184f6af8fb8484959967723e90:526:11\n_callMutatorMethod@http://localhost:12345/packages/allow-deny.js?hash=000d79baa5a670184f6af8fb8484959967723e90:453:27\nupdate@http://localhost:12345/packages/mongo.js?hash=cbccddce42305ea487784467a596d86dc0f6590a:671:21\nupdateLastVersionFlags@http://localhost:12345/app/app.js?hash=54b2eb7ef4304eab89de17962f040d364932a96d:7784:13\ndocuments.insert@http://localhost:12345/app/app.js?hash=54b2eb7ef4304eab89de17962f040d364932a96d:7871:29\nasync*apply/stubReturnValue<@http://localhost:12345/packages/ddp-client.js?hash=9e4081c1b0dfde446688af96b8d9e8494e67fe2a:972:25\nEVp.withValue@http://localhost:12345/packages/meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:1207:15\napply@http://localhost:12345/packages/ddp-client.js?hash=9e4081c1b0dfde446688af96b8d9e8494e67fe2a:963:56\ncall@http://localhost:12345/packages/ddp-client.js?hash=9e4081c1b0dfde446688af96b8d9e8494e67fe2a:863:17\ninsertDocument/<@http://localhost:12345/app/app.js?hash=54b2eb7ef4304eab89de17962f040d364932a96d:9739:12\nEventListener.handleEvent*EventTarget.prototype.addEventListener</<@eval:4:42\ninsertDocument@http://localhost:12345/app/app.js?hash=54b2eb7ef4304eab89de17962f040d364932a96d:9732:10\nonImport@/imports/ui/documents/HealthOneDocumentImportButton.js:42:21\nonChange@/imports/ui/documents/DocumentImportButton.js:72:13\ncallCallback@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:8601:14\ninvokeGuardedCallbackDev@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:8650:16\ninvokeGuardedCallback@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:8712:31\ninvokeGuardedCallbackAndCatchFirstError@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:8726:25\nexecuteDispatch@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:12899:42\nprocessDispatchQueueItemsInOrder@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:12931:22\nprocessDispatchQueue@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:12944:37\ndispatchEventsForPlugins@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:12955:23\ndispatchEventForPluginEventSystem/<@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:13164:12\nbatchedEventUpdates$1@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:27052:12\nbatchedEventUpdates@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:8401:12\ndispatchEventForPluginEventSystem@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:13163:22\nattemptToDispatchEvent@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:10661:36\ndispatchEvent@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:10580:41\nunstable_runWithPriority@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:31640:12\nrunWithPriority$1@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:15932:10\ndiscreteUpdates$1@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:27069:14\ndiscreteUpdates@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:8412:12\ndispatchDiscreteEvent@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:10545:18\nEventListener.handleEvent*EventTarget.prototype.addEventListener</<@eval:4:42\naddEventBubbleListener@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:10667:10\naddTrappedEventListener@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:13067:29\nlistenToNativeEvent@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:13030:28\nlistenToAllSupportedEvents/<@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:12982:28\nlistenToAllSupportedEvents@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:12980:21\ncreateRootImpl@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:30543:31\nReactDOMBlockingRoot@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:30491:24\ncreateLegacyRoot@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:30556:10\nlegacyCreateRootFromDOMContainer@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:30637:10\nlegacyRenderSubtreeIntoContainer@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:30663:44\nrender@http://localhost:12345/packages/modules.js?hash=e1bacdabebfc206afe03ed0e82f9dc2ff00e83d7:30759:10\nmodule/<@http://localhost:12345/app/app.js?hash=54b2eb7ef4304eab89de17962f040d364932a96d:10169:28\nmaybeReady@http://localhost:12345/packages/meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:938:28\nloadingCompleted@http://localhost:12345/packages/meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:950:5\nEventListener.handleEvent*EventTarget.prototype.addEventListener</<@eval:4:42\n@http://localhost:12345/packages/meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:965:14\n@http://localhost:12345/packages/meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:1004:4\n@http://localhost:12345/packages/meteor.js?hash=857dafb4b9dff17e29ed8498a22ea5b1a3d6b41d:1360:3\n"
​
<prototype>: Object { constructor: errorClass(), clone: clone(), stack: "" }
documents.js:237:15
```

**Comment by aureooms** [comment]
Note that the only part of the code that should cause this is [this call to `updateLastVersionFlags`](https://github.com/infoderm/patients/blob/74bf84efc45ddafb7214fce92923c1974677474d/imports/api/documents.js#L314) which potentially updates multiple documents.

**Comment by aureooms** [comment]
NB: the relevant commit is 3b210b1f4a77a107c6306c1ea3935f7caf2a891c.

**Comment by make-github-pseudonymous-again** [comment]
This was fixed in 2aac9cfc0.
